src/axom/sidre/core/Group.hpp

```diff
@@ -154,6 +154,38 @@ class Group
    */
   char getPathDelimiter() const { return s_path_delimiter; }
 
+  /*!
+   * \brief static method to get valid protocols for Group I/O methods.
+   *
+   * Only protocols that work for both input and output are returned.
+   */
+  static const std::vector<std::string>& getValidIOProtocols()
+  {
+    // The first time this is called, fill the vector with valid protocols.
+    if(s_io_protocols.empty())
+    {
+#ifdef AXOM_USE_HDF5
+      s_io_protocols.push_back("sidre_hdf5");
+      s_io_protocols.push_back("conduit_hdf5");
+#endif
+      s_io_protocols.push_back("sidre_json");
+      s_io_protocols.push_back("sidre_conduit_json");
+      s_io_protocols.push_back("conduit_bin");
+      s_io_protocols.push_back("conduit_json");
+      s_io_protocols.push_back("json");
+    }
+
+    return s_io_protocols;
+  }
+
+  /*!
+   * \brief static method to get the default I/O protocol.
+   */
+  static const std::string& getDefaultIOProtocol()
+  {
+    return s_default_protocol;
```

Comment:
Suggestion: It might be clearer if this function defined the default protocol, rather than the `SIDRE_DEFAULT_PROTOCOL` macro/define above.